llvm/test/SYCLLowerIR/esimd_lower_ldst.ll

```diff
@@ -0,0 +1,32 @@
+; RUN: opt -ESIMDLowerLoadStore -S < %s | FileCheck %s
+
+%"class._ZTSN4sycl5intel3gpu4simdIiLi16EEE.sycl::intel::gpu::simd" = type { <16 x i32> }
+
+@vg = dso_local global %"class._ZTSN4sycl5intel3gpu4simdIiLi16EEE.sycl::intel::gpu::simd" zeroinitializer, align 64 #0
+@vc = dso_local addrspace(1) global <16 x i32> zeroinitializer, align 64
+
+; Function Attrs: norecurse nounwind
+define dso_local spir_func void @_Z3foov() local_unnamed_addr #1 {
+; CHECK-LABEL: @_Z3foov(
+; CHECK-NEXT: [[TMP1:%.*]] = call <16 x i32> @llvm.genx.vload.v16i32.p0v16i32(<16 x i32>* getelementptr inbounds (%"class._ZTSN4sycl5intel3gpu4simdIiLi16EEE.sycl::intel::gpu::simd", %"class._ZTSN4sycl5intel3gpu4simdIiLi16EEE.sycl::intel::gpu::simd"* @vg, i64 0, i32 0))
```

Comment:
Is it expected that vload is not lowered in this case? It is not obvious why only vstore is lowered no the normal store.

Reply:
Yes, this is we are testing. vload/vstore associated with global variable with special attritube stays as vload/vstore.  vload/vstore not associated with those global variables is changed to generic load/store. This is explained in the code comment

Reply:
This means that the lowering of the load is not tested, please add additional test to check lowering of the load.

Reply:
load/store are processed in the same fashion. I added one case for load and one case for store. That should be sufficient

Reply:
I don't see the test to check lowering of the load. How I should know if your code works correctly for the case when lowering of the load should happen?

Reply:
I don't see a test for load lowering, how I could be sure that your code for load lowering is working?